core/templates/dev/head/pages/library/SearchBarDirective.js

```diff
@@ -43,6 +43,63 @@ oppia.directive('searchBar', [
               }
             )
           );
+          $scope.activeMenuName = '';
+          $scope.ACTION_OPEN = 'open';
+          $scope.ACTION_CLOSE = 'close';
+          $scope.KEYBOARD_EVENT_TO_KEY_CODES = {
+            enter: {
+              shiftKeyIsPressed: false,
+              keyCode: 13
+            },
+            tab: {
+              shiftKeyIsPressed: false,
+              keyCode: 9
+            },
+            shiftTab: {
+              shiftKeyIsPressed: true,
+              keyCode: 9
+            }
+          };
+          /**
+           * Opens the submenu.
+           * @param {object} evt
+           * @param {String} menuName - name of menu, on which
+           * open/close action to be performed (category,language).
+           */
+          $scope.openSubmenu = function(evt, menuName) {
+            // Focus on the current target before opening its submenu.
+            angular.element(evt.currentTarget).focus();
+            $scope.activeMenuName = menuName;
+          };
+          /**
+           * Handles keydown events on menus.
+           * @param {object} evt
+           * @param {String} menuName - name of menu to perform action
+           * on(category/language)
+           * @param {object} eventsTobeHandled - Map keyboard events('Enter') to
+           * corresponding actions to be performed(open/close).
+           *
+           * @example
+           *  onMenuKeypress($event, 'category', {enter: 'open'})
+           */
+          $scope.onMenuKeypress = function(evt, menuName, eventsTobeHandled) {
+            var targetEvents = Object.keys(eventsTobeHandled);
+            for (var i = 0; i < targetEvents.length; i++) {
+              var keyCodeSpec =
+                $scope.KEYBOARD_EVENT_TO_KEY_CODES[targetEvents[i]];
+              if (keyCodeSpec.keyCode === evt.keyCode &&
+                evt.shiftKey === keyCodeSpec.shiftKeyIsPressed) {
+                if (eventsTobeHandled[targetEvents[i]] === $scope.ACTION_OPEN) {
+                  $scope.openSubmenu(evt, menuName);
+                } else if (eventsTobeHandled[targetEvents[i]] ===
+                  $scope.ACTION_CLOSE) {
+                  $scope.closeSubmenu(evt);
```

Comment:
Where is this defined now? 

(As I said earlier, please don't just remove functions. Check why are they used, do we need it here and then remove if they are not needed.)

Reply:
Got it.